data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/admin/KafkaAdminAccessor.java

```diff
@@ -0,0 +1,103 @@
+package org.opensearch.dataprepper.plugins.kafka.admin;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.KafkaAdminClient;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsResult;
+import org.apache.kafka.clients.admin.ListOffsetsResult;
+import org.apache.kafka.clients.admin.OffsetSpec;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.opensearch.dataprepper.plugins.kafka.consumer.TopicEmptinessMetadata;
+import org.opensearch.dataprepper.plugins.kafka.util.KafkaClusterAuthConfig;
+import org.opensearch.dataprepper.plugins.kafka.util.KafkaSecurityConfigurer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+public class KafkaAdminAccessor {
+    static final Logger LOG = LoggerFactory.getLogger(KafkaAdminAccessor.class);
+
+    private final AdminClient kafkaAdminClient;
+    private final TopicEmptinessMetadata topicEmptinessMetadata;
+    private final List<String> consumerGroupIds;
+
+    public KafkaAdminAccessor(final KafkaClusterAuthConfig kafkaClusterAuthConfig, final List<String> consumerGroupIds) {
+        Properties authProperties = new Properties();
+        KafkaSecurityConfigurer.setAuthProperties(authProperties, kafkaClusterAuthConfig, LOG);
+        this.kafkaAdminClient = KafkaAdminClient.create(authProperties);
+        this.topicEmptinessMetadata = new TopicEmptinessMetadata();
+        this.consumerGroupIds = consumerGroupIds;
+    }
+
+    @VisibleForTesting
+    KafkaAdminAccessor(final AdminClient kafkaAdminClient, final TopicEmptinessMetadata topicEmptinessMetadata, final List<String> consumerGroupIds) {
+        this.kafkaAdminClient = kafkaAdminClient;
+        this.topicEmptinessMetadata = topicEmptinessMetadata;
+        this.consumerGroupIds = consumerGroupIds;
+    }
+
+    public synchronized boolean areTopicsEmpty() {
```

Comment:
minor: From reusability point of view, this can be renamed to something like get consumerLags (topic+consumeGroup) and return per partition lag. then caller or another APi can decide if topic is empty if lags=0. non-blocking can be changed later also.